src/opm/parser/eclipse/EclipseState/Schedule/MSW/Segment.cpp

```diff
@@ -152,5 +153,63 @@ namespace Opm {
     }
 
 
+    // TODO: we can pass in the segment length here. It is tricky anyway
+    void Segment::updateValve(const Valve& valve, const double segment_length) {
+        // TODO: assert here is for debugging purpose, will be improved later
+        // we need to update some values for the vale
+        auto valve_ptr = std::make_shared<Valve>(valve);
+
+        if (valve_ptr->pipeAdditionalLength() < 0.) { // defaulted for this
+            valve_ptr->setPipeAdditionalLength(segment_length);
+        }
+
+        if (valve_ptr->pipeDiameter() < 0.) {
+            valve_ptr->setPipeDiameter(m_internal_diameter);
+            if (this->m_valve != nullptr) {
+                valve_ptr->setPipeDiameter(this->m_valve->pipeDiameter());
```

Comment:
This line looks like it can be removed. Also lines 180 and 191.

Reply:
They can not. Valve_ptr is different from this->m_valve at the moment.